ietf-layer0-types.yang

```diff
@@ -892,6 +892,230 @@ module ietf-layer0-types {
+          description
+            "Channel wavelength computing input.";
+          reference
+            "RFC 6205: Generalized Labels for Lambda-Switch-Capable
+             (LSC) Label Switching Routers";
+        }
+      }
+      case flexi-grid {
+        uses l0-types:flexi-grid-label-start-end;
+      }
+    }
+    reference
+      "RFC 6205: Generalized Labels for Lambda-Switch-Capable (LSC)
+       Label Switching Routers";
+  }
+
+  grouping wdm-label-step {
+    description
+      "Label step information for fixed & flexi-DWDM or CWDM grid";
+    choice l0-grid-type {
+      description
+        "Grid type: DWDM, CWDM, etc.";
+      case fixed-dwdm {
+        leaf wson-dwdm-channel-spacing {
+          when "derived-from-or-self(../../grid-type,
+                \"wson-grid-dwdm\")" {
+            description
+              "Valid only when grid type is DWDM.";
+          }
+          type identityref {
+            base dwdm-ch-spc-type;
+          }
+          description
+            "Label-step is the channel spacing (GHz), e.g., 100.000,
+             50.000, 25.000, or 12.500 GHz for DWDM.";
+          reference
+            "RFC 6205: Generalized Labels for Lambda-Switch-Capable
+             (LSC) Label Switching Routers";
+        }
+      }
+      case cwdm {
+        leaf wson-cwdm-channel-spacing {
+          when "derived-from-or-self(../../grid-type,
+                \"wson-grid-cwdm\")" {
+            description
+              "Valid only when grid type is CWDM.";
+          }
+          type identityref {
+            base cwdm-ch-spc-type;
+          }
+          description
+            "Label-step is the channel spacing (nm), i.e., 20 nm
+             for CWDM, which is the only value defined for CWDM.";
+          reference
+            "RFC 6205: Generalized Labels for Lambda-Switch-Capable
+             (LSC) Label Switching Routers";
+        }
+      }
+      case flexi-grid {
+        uses flexi-grid-label-step;
+      }
+    }
+    reference
+      "RFC 6205: Generalized Labels for Lambda-Switch-Capable (LSC)
+       Label Switching Routers,
+       ITU-T G.694.2 (12/2003): Spectral grids for WDM applications:
+       CWDM wavelength grid,
+       RFC 8363: GMPLS OSPF-TE Extensions in Support of Flexi-Grid
+       Dense Wavelength Division Multiplexing (DWDM) Networks";
+  }
+
+  grouping wdm-label-hop {
+    description
+      "Generic label-hop information for fixed & flexi-DWDM or
+       CWDM grid";
+    choice grid-type {
+      description
+        "Label for DWDM or CWDM grid";
+      case fixed-dwdm {
+        choice fixed-single-or-super-channel {
+          description
+            "single or super channel";
+          case single {
+            leaf dwdm-n {
+              type l0-types:dwdm-n;
+              description
+                "The given value 'N' is used to determine the
+                 nominal central frequency.";
+            }
+          }
+          case super {
```

Comment:
in case of flexible grid, the "super" choice has been deprecated to use multi
case multi {
        container frequency-slots {
          description
            "The top level container for the list of frequency 
            slots used for flexi-grid super channel.";
          list frequency-slot {
            key "flexi-n";
            min-elements 2;
            uses flexi-grid-frequency-slot;
            description
              "List of frequency slots used for flexi-grid super 
              channel.";
          }
This modifcation have been done to avoid the usage of super field also in case the superchannel had only 1 carrier. Here the min in the list is 2.

Reply:
For fixed grid, only the central frequency N needs to be specified.
Therefore, how about the following changes to fixed-dwdm to align:
- change the case name from "super" to "multi"
- add the min-elements similar to flexi-grid
~~~~
          case multi {
            leaf-list subcarrier-dwdm-n {
              type l0-types:dwdm-n;
              min-elements 2;
              description
                "The given values 'N' are used to determine the
                 nominal central frequency for each subcarrier
                 channel.";
              reference
                "ITU-T Recommendation G.694.1: Spectral grids for
                 WDM applications: DWDM frequency grid";
            }
          }
~~~~